icefoganalytics/travel-authorization
# Ytgov-118: Travel Desk Send to Traveler and Close Button Does Not Workhides Full Travel Auth

Fixes https://github.com/ytgov/travel-authorization/issues/118

# Context

> **Describe the bug** Pre Condition:
> 
>     * Approved Travel Auth
> 
>     * Submitted Travel request from Traveler
>       As the travel desk user I review the travel desk request and add a couple questions to the request in hopes to clarify some items on the travelers request.  I add the questions using the add  question button on the right.  Once questions are add I select Send to traveler on the bottom far right. No visual cue that something happen so I close the pop-up using close on the bottom left.  There is not request remaining on the travel desk grid screen or if I navigate back to the my-travel-requests screen the original approval is gone.  When I try to enter a new approval to reproduce the bug I get a foreign key violation.
> 
> 
> **To Reproduce** Precondition:
> 
>     * Approved Travel Auth
> 
>     * Submitted Travel Desk Request
> 
>     * must be travel desk user
>       Steps to reproduce the behavior:
> 
> 
>     1. Go to '(https://travel-auth-dev.ynet.gov.yk.ca/travel-desk)'
> 
>     2. Select a travel desk request
> 
>     3. Add questions on the right of the pop-up
> 
>     4. Select Send to traveler button on right
> 
>     5. Select close on the left
> 
> 
> **Expected behavior** Send to traveler button will add a notification to the my-travel-request (and an email to the requestor not sure if this has been built) saying that clarification questions are required. The request will also be available on the travel-desk screen with a status of "Pending Answers". The close button should just close the pop-up.
> 
> **Screenshots** Unable to provide screen shots as I a can no longer create any preconditions
> 
> **Desktop (please complete the following information):**
> 
>     * Browser Chrome
> 
>     * Version v2024.7.29.1
> 
> 
> **Additional context** I am not sure if the send to traveler button or the if the close button generated the issue

# Implementation

Fix various travel desk router crash conditions.
Migrate travel desk status prettification to use i18n.
Unbreak various travel desk save failure conditions in the front-end.
Add new "options_provided" page stub for My Travel Request workflow when travel desk request is in "options_provided" state

# Screenshots

![image](https://github.com/user-attachments/assets/c7172372-e8b9-4361-8439-e3ba730a8aa6)
![image](https://github.com/user-attachments/assets/e53ea7b6-0a1f-4714-adbe-1de9674c5404)

# Testing Instructions

1. Run the test suite via dev test (or dev test_api)
2. Boot the app via dev up
3. Log in to the app at http://localhost:8080
4. Select "Administration" from the top right kebab menu.
5. Click on the "User Management" item in the far left.
6. Click on your user to go to the "User Editor" and edit the user.
7. Add the "Travel Desk User" role your user and save.
8. Note that the full app refreshes because you edited the current user. This will not happen if you edit a user who is not the current user.
9. Select "My Travel Requests" from the top dropdown menu.
10. Create a new travel authorization via the "+ Travel Authorization" button.
11. Fill in the "Purpose" and "Name of meeting ..." fields in the "Purpose" section.
12. Fill in the "From", "To", and "Date" fields in the "Details" section.
13. Click the "Generate Estimate" button in the "Approvals" section.
14. Click the "Submit" button in the "Approvals" section.
15. Fill the "Submit to" field with your email, and click the "Submit to Supervisor" button.
16. In the top nav, select "Manager View".
17. Find your travel authorization in the top left "Pending Approvals" panel and click on it.
18. Scroll down to the bottom and click the "Approve" button. It will take a while, because it is trying to look up the submitting user in YG's employee directory.
19. In the top nav, select "My Travel Requests".
20. Find your travel authorization, and click the "Submit Travel Desk Request" button.
21. Fill in the various required fields.
22. Add a flight request.
23. Scroll down to the buttom and click "Submit".
24. In the top nav, select "Travel Desk".
25. Find your travel request and click the "pencil" icon on the far right.
26. Scroll to the bottom and click the "Save Draft" button. Check that this does not crash, and shows a nice popup message.
27. Scroll back to the top, and find the "Add Question" button. Click it.
28. Fill in the question fields and click "Save Draft" again. Check that this does not crash, and shows a nice popup message.
29. Scroll to the bottom and click the "Send to Traveler" button.
30. In the top nav, select "My Travel Requests".
31. Find your travel authorization, and note that it has a new call-to-action type of "Rank Travel Desk Options". Click on the buttom.
32. Check that you are sent to a new page stub, that needs to be built.
33. Check that the "Request" tab link is for the current location.
